### ISSUE TITLE
MixedAir Issues to Explore:  OA flow> Supply flow, economizer controls (CR #7652)

### ISSUE THREAD

**Issue by axelstudios** [NREL/EnergyPlus#2433]
## 
#### Description

MixedAir.f90 calculations need scrutiny to examine and resolve the following issues:
1.  It is possible for minimum OA to exceed the supply air flow rate.  (see CR 7648 and defect file 7648*.idf)
2.  FixedMinimum vs. ProportionalMinimum--related code is commented out. verify if feature working and clean up code
3.  Economizer control issues identified prior to release of V3.0.  Run away volumes of OA are possible with   FixedDryBulb economizer type and values for Economizer Maximum Limit Dry-Bulb Temperature that are high relative to return air temps.  (but errors go away with DifferentialDryBulb); see defect file 7652_FixedDryBulb_Economizer.idf (run in Phoenix TMY2).  May just need to clarify documentation on what FixedDryBulb is supposed to do.  Before we realized that DifferentialDryBulb was the way to go, BG had tried the following code change with some success, but it was rejected by RR prior to release (also too late in the cycle and needing more attention).  

Line 2726 in MixedAir.f90  should (maybe) be:

  OutAirSignal = Min(Max(OutAirSignal, OutAirMinFrac), 1.0D0)    ! BG attempt 11/13/2008 

[WFB 17Mar2009]
1) The variable AirLoopFlow()%MinOutAir (which triggers the error message and is reported in the error message)
is not the actual outdoor air flow rate, it is the flow limit which may be overridden by the actual supply air
flow rate. I added the actual outdoor air flow rate (from the outdoor air node) and the mixed air flow rate to
the diagnostic output of the error message. The actual outdoor air flow rate never exceeds the mixed air
flow rate:

Program Version,EnergyPlus, Version 3.1,IDD_Version 3.1.0.016
   *\* Warning *\* Version: in IDF="3.0.0" not the same as expected="3.1.0"
   ************\* Beginning Zone Sizing Calculations
   ************\* Beginning System Sizing Calculations
   ************\* Beginning Plant Sizing Calculations
   ************\* Testing Individual Branch Integrity
   ************\* All Branches passed integrity testing
   ************\* Testing Individual Supply Air Path Integrity
   ************\* All Supply Air Paths passed integrity testing
   ************\* Testing Individual Return Air Path Integrity
   ************\* All Return Air Paths passed integrity testing
   ************\* No node connection errors were found.
   *\* Warning *\* Output:Table:Monthly requested with SimulationControl Run Simulation for Weather File Run Periods set to No so Output:Table:Monthly will not be generated
   *\* Warning *\* CoolingTower:SingleSpeed "TOWERWATERSYS COOLTOWER"
   *\*   ~~~   *\* Cooling tower water outlet temperature (5.00 C) is below the specified minimum condenser loop temp of 5.00 C
   *\*   ~~~   *\*  Occurrence info=CHICAGO ANN HTG 99.6% CONDNS DB, 01/21 05:20 - 05:25
   *\* Warning *\* SimHVAC: Maximum iterations (25) exceeded for all HVAC loops, at CHICAGO ANN CLG .4% CONDNS DB=>MWB, 07/21 05:00 - 05:15
   *\*   ~~~   *\* ...use Output:Diagnostics,DisplayExtraWarnings;   to show more details on each max iteration exceeded.
   *\* Severe  *\* SimHVAC: Maximum iterations diagnostic. Detected minimum outdoor air flow rate larger than supply air flow rate
   *\*   ~~~   *\* The AirLoopHVAC object is named = VAV_5
   *\*   ~~~   *\* The Controller:OutdoorAir object is named = VAV_5_OA_CONTROLLER
   *\*   ~~~   *\* The minimum outdoor air flow rate is = 1.1285 [kg/s]
   *\*   ~~~   *\* The supply air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The actual outdoor air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The mixed air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* Check minimum air flow fractions for any VAV air terminals connected to this air loop.
   *\*   ~~~   *\* Flow fractions need to exceed outdoor air ratio at all times.
   *\*   ~~~   *\* Outdoor air ratio at time of error = 0.132
   *\* Severe  *\* SimHVAC: Maximum iterations diagnostic. Detected minimum outdoor air flow rate larger than supply air flow rate
   *\*   ~~~   *\* The AirLoopHVAC object is named = VAV_1
   *\*   ~~~   *\* The Controller:OutdoorAir object is named = VAV_1_OA_CONTROLLER
   *\*   ~~~   *\* The minimum outdoor air flow rate is = 2.2570 [kg/s]
   *\*   ~~~   *\* The supply air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The actual outdoor air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The mixed air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* Check minimum air flow fractions for any VAV air terminals connected to this air loop.
   *\*   ~~~   *\* Flow fractions need to exceed outdoor air ratio at all times.
   *\*   ~~~   *\* Outdoor air ratio at time of error = 0.104
   *\* Severe  *\* SimHVAC: Maximum iterations diagnostic. Detected minimum outdoor air flow rate larger than supply air flow rate
   *\*   ~~~   *\* The AirLoopHVAC object is named = VAV_2
   *\*   ~~~   *\* The Controller:OutdoorAir object is named = VAV_2_OA_CONTROLLER
   *\*   ~~~   *\* The minimum outdoor air flow rate is = 22.5706 [kg/s]
   *\*   ~~~   *\* The supply air flow rate is = 22.5241 [kg/s]
   *\*   ~~~   *\* The actual outdoor air flow rate is = 22.5252 [kg/s]
   *\*   ~~~   *\* The mixed air flow rate is = 22.5252 [kg/s]
   *\*   ~~~   *\* Check minimum air flow fractions for any VAV air terminals connected to this air loop.
   *\*   ~~~   *\* Flow fractions need to exceed outdoor air ratio at all times.
   *\*   ~~~   *\* Outdoor air ratio at time of error = 9.688E-002
   *\* Severe  *\* SimHVAC: Maximum iterations diagnostic. Detected minimum outdoor air flow rate larger than supply air flow rate
   *\*   ~~~   *\* The AirLoopHVAC object is named = VAV_3
   *\*   ~~~   *\* The Controller:OutdoorAir object is named = VAV_3_OA_CONTROLLER
   *\*   ~~~   *\* The minimum outdoor air flow rate is = 2.2570 [kg/s]
   *\*   ~~~   *\* The supply air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The actual outdoor air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The mixed air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* Check minimum air flow fractions for any VAV air terminals connected to this air loop.
   *\*   ~~~   *\* Flow fractions need to exceed outdoor air ratio at all times.
   *\*   ~~~   *\* Outdoor air ratio at time of error = 9.321E-002
   *\* Warning *\* SimHVAC: Maximum iterations (25) exceeded for all HVAC loops, at CHICAGO ANN CLG .4% CONDNS DB=>MWB, 07/21 05:15 - 05:22
   *\* Severe  *\* SimHVAC: Maximum iterations diagnostic. Detected minimum outdoor air flow rate larger than supply air flow rate
   *\*   ~~~   *\* The AirLoopHVAC object is named = VAV_5
   *\*   ~~~   *\* The Controller:OutdoorAir object is named = VAV_5_OA_CONTROLLER
   *\*   ~~~   *\* The minimum outdoor air flow rate is = 1.1285 [kg/s]
   *\*   ~~~   *\* The supply air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The actual outdoor air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The mixed air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* Check minimum air flow fractions for any VAV air terminals connected to this air loop.
   *\*   ~~~   *\* Flow fractions need to exceed outdoor air ratio at all times.
   *\*   ~~~   *\* Outdoor air ratio at time of error = 0.132
   *\* Severe  *\* SimHVAC: Maximum iterations diagnostic. Detected minimum outdoor air flow rate larger than supply air flow rate
   *\*   ~~~   *\* The AirLoopHVAC object is named = VAV_1
   *\*   ~~~   *\* The Controller:OutdoorAir object is named = VAV_1_OA_CONTROLLER
   *\*   ~~~   *\* The minimum outdoor air flow rate is = 2.2570 [kg/s]
   *\*   ~~~   *\* The supply air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The actual outdoor air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The mixed air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* Check minimum air flow fractions for any VAV air terminals connected to this air loop.
   *\*   ~~~   *\* Flow fractions need to exceed outdoor air ratio at all times.
   *\*   ~~~   *\* Outdoor air ratio at time of error = 0.104
   *\* Severe  *\* SimHVAC: Maximum iterations diagnostic. Detected minimum outdoor air flow rate larger than supply air flow rate
   *\*   ~~~   *\* The AirLoopHVAC object is named = VAV_2
   *\*   ~~~   *\* The Controller:OutdoorAir object is named = VAV_2_OA_CONTROLLER
   *\*   ~~~   *\* The minimum outdoor air flow rate is = 22.5706 [kg/s]
   *\*   ~~~   *\* The supply air flow rate is = 16.5221 [kg/s]
   *\*   ~~~   *\* The actual outdoor air flow rate is = 16.5231 [kg/s]
   *\*   ~~~   *\* The mixed air flow rate is = 16.5231 [kg/s]
   *\*   ~~~   *\* Check minimum air flow fractions for any VAV air terminals connected to this air loop.
   *\*   ~~~   *\* Flow fractions need to exceed outdoor air ratio at all times.
   *\*   ~~~   *\* Outdoor air ratio at time of error = 9.688E-002
   *\* Severe  *\* SimHVAC: Maximum iterations diagnostic. Detected minimum outdoor air flow rate larger than supply air flow rate
   *\*   ~~~   *\* The AirLoopHVAC object is named = VAV_3
   *\*   ~~~   *\* The Controller:OutdoorAir object is named = VAV_3_OA_CONTROLLER
   *\*   ~~~   *\* The minimum outdoor air flow rate is = 2.2570 [kg/s]
   *\*   ~~~   *\* The supply air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The actual outdoor air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* The mixed air flow rate is = 0.0000 [kg/s]
   *\*   ~~~   *\* Check minimum air flow fractions for any VAV air terminals connected to this air loop.
   *\*   ~~~   *\* Flow fractions need to exceed outdoor air ratio at all times.
   *\*   ~~~   *\* Outdoor air ratio at time of error = 9.321E-002

Thus there does not seem to be any air flow mass balance problem here. In fact, this problem (max iters)
has nothing to do with OA or min OA. When min OA is lowered to almost zero, the problem persists. 
It is a low flow problem however, and the problem is coming from the fan heat to airstream. At low
flow this heat can cause a large temperature rise and produce air loop solution oscillations. When we run
with the fan min flow set to 0.01 m3/s, the problem goes away.

2) I have checked thatFixedMinimum vs. ProportionalMinimum  is working. The commented out code is not
 needed. The min OA frac is calculated for  ProportionalMinimum. Then this code deals with FixedMinimum:

! if fixed minimum, don't let go below min OA
IF (OAController(OAControllerNum)%FixedMin) THEN
! cycling fans allow "average" min OA to be below minimum
  IF (.NOT. AirLoopCyclingFan) THEN
    OAController(OAControllerNum)%OAMassFlow = MAX(OAController(OAControllerNum)%OAMassFlow,&
                                                   OAController(OAControllerNum)%MinOAMassFlowRate \* MinOASchedVal)
  END IF
END IF

3) FixedDryBulb is working as intended. There is no " Run away volumes of OA". The controller is doing
what it is told to do and can not intuit the user's intent. I believe the situation is well described in the
IO Ref:

The outdoor air controller includes a number of user-selectable limit controls. If any
of the selected limits are exceeded, the outdoor airflow rate is set to the minimum.

If all the limits are satisfied, the outdoor air controller does the following for continuous air
flow systems: if the outdoor air temperature is greater than or equal to the mixed air
temperature setpoint, the outdoor air flow rate is set to the maximum; if the outdoor air
temperature is less than the mixed air temperature setpoint, the outdoor air controller will
modulate the outdoor air flow so that the mixed air temperature will match the mixed air
setpoint temperature.

Choosing FixedDryBulb means the economizer will set the outdoor airflow rate at minimum if
the outdoor air temperature is higher than a specified dry-bulb temperature limit.
# 

In summary, there is no OA or min OA issue. The new warning messages should probably be removed.
There is a low flow instability problem coming from the fan heat. I am marking this CR "as designed" and
adding a new enhancement list item HVAC_AirLoop_2009_21 that describes the fan heat problem.

[BG 3-17-09] marking as verified. Going to change HVACMAnager.f90 to comment out call to DiagnoseHVACMaxIterationError with new warning message per discussion above.
## 
#### Workaround

Reduce the fan minimum flow input item to a small (.01 m3/s for instance) number. Autosizing is not
getting a good number for this.
## 

Addressed by F Buhl
Last build tested: `08.12.10 V3.1.0.003`
